src/common/types/vector.cpp

```diff
@@ -2434,6 +2434,23 @@ const vector<unique_ptr<Vector>> &StructVector::GetEntries(const Vector &vector)
 	return GetEntries((Vector &)vector);
 }
 
+vector<string> StructVector::GetKeys(Vector &vector) {
```

Comment:
I wonder if we need this function. We could also just use `StructType::GetChildTypes` in your scalar function code directly, and change the loop there. That way, we only iterate over the child types once. That might be a good optimization for bigger structs.

```cpp
	for (idx_t i = 0; i < key_count; i++) {
// here, instead of `keys`, we can loop over `child_types`, and pass `child_type.first`.
		child_data[i] = StringVector::AddString(list_child, keys[i]);
	}
```